pkg/genericapiserver/storage_factory.go

```diff
@@ -88,6 +88,11 @@ type groupResourceOverrides struct {
 	// of exposing one set of concepts.  autoscaling.HPA and extensions.HPA as a for instance
 	// The order of the slice matters!  It is the priority order of lookup for finding a storage location
 	cohabitatingResources []unversioned.GroupResource
+
+	// ignoreCohabitatingStorageVersion controls whether or not a cohabitating resource attempts to serialize into the same
+	// groupVersion as all other cohabitators.  If not, it uses the groupVersion assigned to itself instead of the one being
+	// used by the first enabled cohabitator.
+	ignoreCohabitatingStorageVersion bool
```

Comment:
I don't really see why-- suppose we expose this in the discovery as a
"storage key collection" or some such. I don't see why my suggested way
would produce any different discovery documents?

On Fri, Jul 22, 2016 at 11:24 AM, David Eads <notifications@github.com>
wrote:

> In pkg/genericapiserver/storage_factory.go
> <https://github.com/kubernetes/kubernetes/pull/29306#discussion_r71923336>
> :
>
> > @@ -88,6 +88,11 @@ type groupResourceOverrides struct {
> >  	// of exposing one set of concepts.  autoscaling.HPA and extensions.HPA as a for instance
> >  	// The order of the slice matters!  It is the priority order of lookup for finding a storage location
> >  	cohabitatingResources []unversioned.GroupResource
> > +
> > +	// ignoreCohabitatingStorageVersion controls whether or not a cohabitating resource attempts to serialize into the same
> > +	// groupVersion as all other cohabitators.  If not, it uses the groupVersion assigned to itself instead of the one being
> > +	// used by the first enabled cohabitator.
> > +	ignoreCohabitatingStorageVersion bool
>
> If we want that, why would we do this rather than just making two separate
> resources that both happen to use the same etcd key?
>
> I think that this structure better describes the intent. These resources
> are cohabiting, they just aren't stored in the same format yet. When we try
> to summarize this information under a discovery endpoint, I think it will
> be easier to harvest that information if we're able to inspect cohabiters
> instead inspecting the different resultant etcd locations and prefixes from
> across the config.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29306/files/43cf3edd2a0830b7365ee9e7b8d4f28a982f6748#r71923336>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnglkkw7skZceYdhYuK5QF_1W23RNqJks5qYQrZgaJpZM4JRELN>
> .
>